### PR TITLE
Encapsulate `validate_header_regarding_parent` inside `SealedHeader`

### PR DESCRIPTION
**Pull request by tcoratger**
### Changes Made
This pull request encapsulates the `validate_header_regarding_parent` function within the `SealedHeader` struct. The encapsulation is done to improve code organization and enhance encapsulation of header-related validation logic.

- Errors from `validate_header_regarding_parent` are mapped inside a `SealedHeaderError` enum.
- An error field `SealedHeaderError` is introduced inside `ConsensusError` enum, transparently wrapping `SealedHeaderError`
- All old unused `ConsensusError` error fields that are wrapped inside `SealedHeaderError` are removed from `ConsensusError` enum to clean up and better organize errors there.

Should close #6193.

### Motivation
Encapsulating `validate_header_regarding_parent` within `SealedHeader` promotes a cleaner code structure by keeping related functionality within the same module. This encapsulation enhances the readability and maintainability of the codebase.

### Impact
This change does not introduce any functional changes but improves the organization of the code. Existing functionality remains unchanged, and this encapsulation serves as a step towards better code organization.

### CODE REVIEW

File: crates/primitives/src/header.rs [hunk 1]
```diff
@@ -574,6 +578,92 @@ impl Decodable for Header {
     }
 }
 
+/// Errors that can occur during header sanity checks.
+#[derive(thiserror::Error, Debug, PartialEq, Eq, Clone)]
+pub enum SealedHeaderError {
```
Comment (mattsse) [inline crates/primitives/src/header.rs:583]: perhaps `HeaderValidationError` ?

File: crates/primitives/src/header.rs [hunk 2]
```diff
@@ -586,6 +676,184 @@ pub struct SealedHeader {
...
+    /// Returns a [`SealedHeaderError`] if any validation check fails, indicating specific issues
+    /// with the sealed header. The possible errors include mismatched block numbers, parent
+    /// hash mismatches, timestamp inconsistencies, gas limit violations, base fee discrepancies
+    /// (for EIP-1559), and errors related to the blob gas fields (EIP-4844).
+    ///
+    /// ## Note
+    ///
+    /// Some checks, such as gas limit validation, are conditionally skipped based on the presence
+    /// of certain features (e.g., Optimism feature) or the activation of specific hardforks.
+    pub fn validate_regarding_parent(
```
Comment (mattsse) [inline crates/primitives/src/header.rs:749]:
```suggestion
    pub fn validate_against_parent(
```

**Review by mattsse — CHANGES_REQUESTED** [review]
I like this a lot, this is much better than having standalone functions.

some naming nits otherwise lgtm

pending @Rjected

File: crates/primitives/src/header.rs (hunk 1 above)
Comment (tcoratger) [reply crates/primitives/src/header.rs]: Done

File: crates/primitives/src/header.rs (hunk 2 above)
Comment (tcoratger) [reply crates/primitives/src/header.rs]: Done

**Review by mattsse — APPROVED** [review]
lgtm, pending @Rjected

**Review by Rjected — APPROVED** [review]
LGTM, I like that this puts the header validation errors in a new type

**Pull request merged**